extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/pages/contract_agreements/ContractAgreementPageApiService.java

```diff
@@ -29,15 +31,25 @@ public class ContractAgreementPageApiService {
     private final ContractAgreementPageCardBuilder contractAgreementPageCardBuilder;
 
     @NotNull
-    public ContractAgreementPage contractAgreementPage() {
+    public ContractAgreementPage contractAgreementPage(@Nullable ContractAgreementPageQuery contractAgreementPageQuery) {
         var agreements = contractAgreementDataFetcher.getContractAgreements();
 
         var cards = agreements.stream()
-                .map(agreement -> contractAgreementPageCardBuilder.buildContractAgreementCard(
-                        agreement.agreement(), agreement.negotiation(), agreement.asset(), agreement.transfers()))
-                .sorted(Comparator.comparing(ContractAgreementCard::getContractSigningDate).reversed())
-                .toList();
+            .map(agreement -> contractAgreementPageCardBuilder.buildContractAgreementCard(
+                agreement.agreement(),
+                agreement.negotiation(),
+                agreement.asset(),
+                agreement.transfers(),
+                agreement.terminations()))
```

Comment:
only 0..1 terminations per agreement, please.

it's "either terminated or not". 

backend-for-frontend, let's not make the UI have to deal with any biz like that